### ISSUE TITLE
Create direct Gradient access for PyTorch

### ISSUE THREAD

**Issue by EricKolibacz** [dailab/MAXi-XAI-lib#10]
We would like to access the Model gradient directly to speed up the explanation generation. First, this should be implemented in PyTorch, TensorFlow or support for other libraries should follow.
